api/scripts/bigint/answer/add-triggers-to-copy-data-from-answer-ke-to-tmp-bigint-tables.js

```diff
@@ -0,0 +1,116 @@
+           SET "endsAtId" = (SELECT MAX(id) FROM "answers")
+           WHERE bms."tableName" = 'answers'`
+      )
+      .transacting(trx);
+
+    await knex
+      .raw(
+        `UPDATE "bigint-migration-settings" bms
+           SET "endsAtId" = (SELECT MAX(id) FROM "knowledge-elements")
+           WHERE bms."tableName" = 'knowledge-elements'`
+      )
+      .transacting(trx);
+
+    await knex
+      .raw(
+        `CREATE OR REPLACE FUNCTION "insert_answers_in_answers_bigint"()
+                    RETURNS TRIGGER AS
+                    $$
+                    BEGIN
+                      INSERT INTO "answers_bigint" ("id", "value", "result", "assessmentId", "challengeId", timeout, "resultDetails", "timeSpent", "isFocusedOut")
+                      VALUES         (NEW."id"::BIGINT, NEW."value", NEW."result", NEW."assessmentId", NEW."challengeId", NEW."timeout", NEW."resultDetails", NEW."timeSpent", NEW."isFocusedOut");
+                     RETURN NEW;
+                    END
+                    $$ LANGUAGE plpgsql;`
+      )
+      .transacting(trx);
+
+    await knex
+      .raw(
+        `CREATE TRIGGER "trg_answers"
+                    AFTER INSERT ON "answers"
+                    FOR EACH ROW
+                    EXECUTE FUNCTION "insert_answers_in_answers_bigint"();`
+      )
+      .transacting(trx);
+
+    await knex
+      .raw(
+        `CREATE OR REPLACE FUNCTION "insert_knowledge-elements_in_knowledge-elements_bigint"()
+                    RETURNS TRIGGER AS
+                    $$
+                    BEGIN
+                       INSERT INTO "knowledge-elements_bigint"( id, source, status, "createdAt", "answerId", "assessmentId", "skillId", "earnedPix", "userId", "competenceId")
+                       VALUES ( NEW.id::BIGINT, NEW.source, NEW.status, NEW."createdAt", NEW."answerId", NEW."assessmentId", NEW."skillId", NEW."earnedPix", NEW."userId", NEW."competenceId");
+                      RETURN NEW;
+                    END
+                    $$ LANGUAGE plpgsql;`
+      )
+      .transacting(trx);
+
+    await knex
+      .raw(
+        `CREATE TRIGGER "trg_knowledge-elements"
+                    AFTER INSERT ON "knowledge-elements"
+                    FOR EACH ROW
+                    EXECUTE FUNCTION "insert_knowledge-elements_in_knowledge-elements_bigint"();`
+      )
+      .transacting(trx);
+  });
+  const endTime = performance.now();
+
+  logger.debug(`Call to addTriggersToAnswersKeBigintTemporaryTables took ${endTime - startTime} milliseconds`);
+  logger.debug('Finish adding triggers');
+};
+
+async function main() {
+  try {
+    logger.debug('Start script... ');
+    await addTriggersToAnswersKeBigintTemporaryTables();
+    logger.debug('End script: done successfully.');
+  } catch (error) {
+    console.error(error);
+    process.exit(1);
+  }
+}
+
+if (require.main === module) {
+  main().then(
+    () => process.exit(0),
+    (err) => {
+      console.error(err);
+      process.exit(1);
```

Comment:
**question**
As-tu essayé de le remplacer par `process.exitCode = 1` ?
https://nodejs.dev/learn/how-to-exit-from-a-nodejs-program

Reply:
Ca revient à la même chose, on garde du coup le process.exit pour être cohérent avec l'existant